pkg/provider/provider.go

```diff
@@ -94,7 +94,8 @@ func (p *KedaProvider) GetExternalMetric(namespace string, metricSelector labels
 			}
 			// Filter only the desired metric
 			if strings.EqualFold(metricSpec.External.Metric.Name, info.Metric) {
-				metrics, err := scaler.GetMetrics(context.TODO(), info.Metric, metricSelector)
+				metrics, err := p.getMetricsWithFallback(scaler, info.Metric, metricSelector, scaledObject, metricSpec)
```

Comment:
Can we call this method only if fallback is enabled? I see there's a `switch` at the end of `getMetricsWithFallback()`, but I'd prefer to not calling this function if it is not needed.

Reply:
I think the issue with this is that we've essentially added functionality to update the scaled object status even when fallback isn't enabled (as discussed [here](https://github.com/kedacore/keda/issues/1872#issuecomment-862937569)). Right now that's neatly isolated away from the "business logic" area of the codebase, which remains here.

It feels like it might obfuscate some existing behaviour if we move that logic into here and also have the status set for fallbacks inside the other file as well. Thoughts about this?

I think that within `getMetricsWithFallback`, we already return as soon as possible; everything that comes before that is the status update logic.